nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java

```diff
@@ -351,6 +372,59 @@ synchronized private void createHttpServerFromService(final ProcessContext conte
         initialized.set(true);
     }
 
+    private ServerConnector createServerConnector(Server server, int port, SSLContextService sslContextService, boolean sslRequired, boolean needClientAuth) {
+        final SslContextFactory contextFactory = createSslContextFactory(sslContextService, sslRequired, needClientAuth);
```

Comment:
I don't think `createServerConnector()` should be overloaded / separated into 2 methods.
This single line can be moved to the next method.
Actually, creating the `SslContextFactory` is only needed in the `if (sslRequired) {}` block and should not be called otherwise.

Reply:
You are right, thanks! The modifications are in the next commit.